src/app/rpmostree-builtin-status.cxx

```diff
@@ -1212,18 +1212,22 @@ print_history_entry (RORHistoryEntry  *entry,
 
   if (!opt_json)
     {
-      print_timestamp_and_relative ("BootTimestamp", entry->last_boot_timestamp);
-      if (entry->boot_count > 1)
+      print_timestamp_and_relative ("BootTimestamp", entry.last_boot_timestamp);
+      if (entry.boot_count > 1)
         {
           g_print ("%s BootCount: %" PRIu64 "; first booted on ",
-                   libsd_special_glyph (TREE_RIGHT), entry->boot_count);
-          print_timestamp_and_relative (NULL, entry->first_boot_timestamp);
+                   libsd_special_glyph (TREE_RIGHT), entry.boot_count);
+          print_timestamp_and_relative (NULL, entry.first_boot_timestamp);
         }
 
-      print_timestamp_and_relative ("CreateTimestamp", entry->deploy_timestamp);
-      if (entry->deploy_cmdline)
-        g_print ("CreateCommand: %s%s%s\n",
-                 get_bold_start (), entry->deploy_cmdline, get_bold_end ());
+      print_timestamp_and_relative ("CreateTimestamp", entry.deploy_timestamp);
+      if (entry.deploy_cmdline.length() > 0)
+        {
+          // Copy so we can use c_str()
+          auto cmdline_copy = std::string(entry.deploy_cmdline);
```

Comment:
This one is actually interesting - it gets to a point @lucab asked about in a team chat.  Basically in C++ `const` is a lot more useful than C, and since our Rust API returns a non-mutable value it shows up on the C++ side as `const`.  And the way `c_str()` is implemented for `rust::String` requires mutating it today.

(At some point again I hope to look at adding `CUtf8Buf` support to cxx-rs or so, but in the short term we already malloc all over the place, and this isn't a hot path)